File: TP1 - Dart/domain/entities/employee.dart
import './entities.dart';

class Employee extends User {
  final String _id;
  TimeSheet _timeSheet;
  List<Payment> _payments;

  Employee(this._id)
      : _timeSheet = TimeSheet(_id),
        _payments = [],
        super(_id);

  // Getter's
  String get id => this._id;
  TimeSheet get timeSheet => this._timeSheet;
  List<Payment> get payments => this._payments;

  // Setter's
  set timeSheet(TimeSheet timeSheet) => this._timeSheet = timeSheet;
  set payments(List<Payment> payments) => this._payments = payments;
}

File: TP2 - Java/src/domain/entities/Employee.java
package domain.entities;

import java.util.Arrays;
import java.util.UUID;

public class Employee extends User {
  private TimeSheet timeSheet;
  private Payment payments[];

  public Employee(String id, String name, String login, String password, UserPermissions userPermissions) {
    super(id, name, login, password, userPermissions);
    this.timeSheet = new TimeSheet(UUID.randomUUID().toString(), Arrays.asList());
  }

  public TimeSheet getTimeSheet() {
    return timeSheet;
  }

  public void setTimeSheet(TimeSheet timeSheet) {
    this.timeSheet = timeSheet;
  }

  public Payment[] getPayments() {
    return payments;
  }

  public void setPayments(Payment[] payments) {
    this.payments = payments;
  }
}
